pkg/command-handler/domain/ledger/entities/account_class_types.go

```diff
@@ -0,0 +1,19 @@
+package entities
+
+// ClassTypes has the set of class types.
+var ClassTypes = AccountClassTypes{
+	"liability": empty,
+	"assets":    empty,
+	"income":    empty,
+	"expense":   empty,
+	"equity":    empty,
+}
+
+type AccountClassTypes map[string]struct{}
+
+var empty struct{}
+
+func (act AccountClassTypes) Has(classType string) bool {
+	_, ok := act[classType]
+	return ok
+}
```

Comment:
Maybe a simple private map and a simple public function will be enough

```suggestion
package entities

// ClassTypes has the set of class types.
var classTypes = map[string]bool{
	"liability": true,
	"assets":    true,
	"income":    true,
	"expense":   true,
	"equity":    true,
}

func HasClassType(classType string) bool {
	_, ok := classTypes[classType]
	return ok
}

```

Reply:
NIT: The current implementation has the same effect as the **SET** (data structure) because it avoids elements duplication and the _empty_ var has no penalty in performance (0 allocations). 

The suggestion is "good" for readability but on the flip side it seems messy for the semantics since the `classTypes` maps **string**s to **true**s.  

I'm not against it(at all) but it should be changed intentionally and not because of code size. Am I losing something?

Reply:
I'm not a Go developer, so I'm not the right person to know which one, struct or bool, is the better solution.
I just don't like to expose mutable(I love immutability) map unnecessarily and will keep the code very simple.

Reply:
In my last commit, the account class is a type. In this way, is possible to create a new instance (from string) and to export the value to string too.

Cam you TAL? @alexrios @robsonpeixoto

Reply:
@alexgarzao LGTM 👍🏾 

Besides I have 2 minors which is a matter of style:

- `NewAccountClassFromString` have no reason to return a pointer, it could be a simple value.
-  We're "eagerly" adding some indexes and no query is making use of it in `where` clauses. It's overheads the writes and have no real gain since an index added after the values generate better b-trees.

Reply:
Thanks for your review :-)

About the NewAccountClassFromString, ok, I will fix it.

About the indexes. These indexes will be used in the next PR. But here, in this PR, you are right, they are useless. I will remove them.

Reply:
@alexrios  About your comment related to returns a simple value. What are "the best practices" to known when must be returned a pointer or a value? My first impression is that return a pointer is more idiomatic.

How do you decide when to return a pointer or a value?

Reply:
It's not a rule and out of scope of this PR but my take on this is asking myself as a client of a function: when I'm going to add or remove something from a value of this type, do I create a new value or change the current value?

An analogy is a function producing a dollar (like a value object). We are not really interested in what (identity) dollar it is (pointers) but in the monetary value (values).

About performance:
- Value as default option because they keep all the data generated on program stack, i.e., no GC involved (less latency).
- Values will probably use more memory for a short period of time in cases the function are return values and errors, but in scale of dozens of bytes.
- Pointers to produce long lifetime structs such as logger and connections, those structs are commonly pictured on project architecture representing usecases, repositories and so on.
- Pointers to very large structs or types. (I'd rather to benchmark first instead of pushing pointers blindly)
- Again, short lifetime objects stress the GC a lot.

All these questions are very minor. I'll never **enforce** this rules if it compromises the product value delivery stream (AKA create tech debt).